bundle/edu.gemini.programexport.servlet/src/main/scala/edu/gemini/programexport/servlet/ProgramExportServlet.scala

```diff
@@ -0,0 +1,232 @@
+
+  def siteQualityFields(s: SPSiteQuality): Json =
+    ("cc" := s.getCloudCover.displayValue) ->:
+      ("sb" := s.getSkyBackground.displayValue) ->:
+      ("iq" := s.getImageQuality.displayValue) ->:
+      ("wv" := s.getWaterVapor.displayValue) ->:
+      ("elevationConstraintType" := s.getElevationConstraintType.displayValue) ->:
+      ("elevationConstraintMin" := s.getElevationConstraintMin) ->:
+      ("elevationConstraintMax" := s.getElevationConstraintMax) ->:
+  // We should probably omit this if there are no timing windows
+      ("timingWindows" := s.getTimingWindows.asScala.map { tw =>
+        ("start" := tw.getStart) ->:
+          ("duration" := tw.getDuration) ->:
+          ("period" := tw.getPeriod) ->:
+          ("repeat" := tw.getRepeat) ->:
+          jEmptyObject
+      }.zipWithIndex.foldLeft(jEmptyObject){ case (j, (tw, idx)) =>
+        (s"timingWindow$idx" := tw) ->: j
+      }) ->:
+      jEmptyObject
+//      ("timingWindows" := s.getTimingWindows.asScala.zipWithIndex.foldLeft(jEmptyObject) { case (j, (tw, idx)) =>
+//        ("windowIndex" := idx) ->: (
+//          ("start" := tw.getStart) ->:
+//            ("duration" := tw.getDuration) ->:
+//            ("period" := tw.getPeriod) ->:
+//            ("repeat" := tw.getRepeat) ->:
+//            j)
+//      }) ->:
+//      jEmptyObject
+
+  def groupFields(g: SPGroup): Json =
+    ("name" := g.getTitle) ->: jEmptyObject
+
+  def observationFields(o: SPObservation): Json = {
+    ("title" := o.getTitle) ->:
+      ("phase2Status" := o.getPhase2Status.displayValue) ->:
+      ("execStatusOverride)" :=? o.getExecStatusOverride.asScalaOpt.map(_.displayValue)) ->?:
+      ("priority" := o.getPriority.displayValue) ->:
+      ("tooOverrideRapid" := o.isOverrideRapidToo.toYesNo.displayValue) ->:
+      // TODO: Unsure if this is the way to handle this data.
+      // TODO: SPObservation only has a setup time type, which is an enum in SetupTime,
+      // TODO: where the Durations are actually stored.
+      // TODO: Apparently, this is in the instruments by the description.
+      ("setupTime" := o.getSetupTimeType.toString) ->:
+      // TODO: Also need acquisition overhead here. Where is this?
+      jEmptyObject
+  }
+
+  def sequenceNode(n: ISPNode): Option[JsonAssoc] =
+    n.dataObject.flatMap {
+      case _: SeqBase => Option(n.getContextObservation).map(sequence)
+      case _          => None
+    }
+
+  def sequence(o: ISPObservation): JsonAssoc = {
+    val steps = ConfigBridge
+      .extractSequence(o, null, ConfigValMapInstances.TO_SEQUENCE_VALUE)
+      .getAllSteps.toList
+
+    "sequence" := steps.map {s =>
+      s.itemEntries.toList.foldLeft(jEmptyObject) { (j, e) =>
+        // TODO: e is an ItemEntry and the value should have been mapped to a String by TO_SEQUENCE_VALUE
+        // TODO: I suspect this has something to do with Scala / Java Option since
+        // TODO: TO_SEQUENCE_VALUE returns a Java Option<String>.
```

Comment:
What is happening?  I'm not sure I understand.  At any rate, the `Option<String>` returned by `TO_SEQUENCE_VALUE.objectToString` is handled internally by `ConfigValMapToString` so it should be getting done correctly or else we'd have heard about it before now (hehe).

Reply:
If I don't put a `toString` here, I seem to be getting a type issue:
```
(e.getKey.getPath := e.getItemValue(?: EncodeJson[AnyRef])) ->: j
```
`e.getItemValue` seems to be of type `AnyRef`?

Reply:
I checked the code base and there don't seem to be any examples of this in Scala, as far as I can tell, unless I missed something, so I'm guessing that that has to do with the `AnyRef` issue.